src/Test.UnitTests.Sarif/Core/SerializedPropertyInfoTests.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Newtonsoft.Json;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Core
+{
+    public class SerializedPropertyInfoTests
+    {
+        [Fact]
+        public void SerializedPropertyInfo_ComparerTests()
+        {
+            Guid testGuid = Guid.NewGuid();
+            DateTime now = DateTime.UtcNow;
+
+            var testCases = new[]
+            {
+                new
+                {
+                    Left = new SerializedPropertyInfo(null, isString:false),
+                    Right = new SerializedPropertyInfo(null, isString:false),
+                    ExpectedEqualsResult = true,
+                    ExpectedCompareResult = 0
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(null, isString:false),
+                    Right = new SerializedPropertyInfo(null, isString:true),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = -1
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(true), isString:false),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(true), isString:false),
+                    ExpectedEqualsResult = true,
+                    ExpectedCompareResult = 0
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(true), isString:false),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(false), isString:false),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = 1
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(10), isString:false),
```

Comment:
>new SerializedPropertyInfo(JsonConvert.SerializeObject(10), isString:false), [](http://example.com/codeflow?start=27&length=76)

some combination of below that you can consider:

new SerializedPropertyInfo(JsonConvert.SerializeObject(10), isString:false),
new SerializedPropertyInfo(JsonConvert.SerializeObject(10), isString:true),
new SerializedPropertyInfo(JsonConvert.SerializeObject("10"), isString:false),
new SerializedPropertyInfo(JsonConvert.SerializeObject("10"), isString:true),
new SerializedPropertyInfo(JsonConvert.SerializeObject("\"10\""), isString:false),
new SerializedPropertyInfo(JsonConvert.SerializeObject("\"10\""), isString:true),

And if user use SetProperty() instead, will there be problem

Reply:
added the test cases